sktime/datatypes/_series/_convert.py

```diff
@@ -114,7 +114,7 @@ def convert_MvS_to_np_as_Series(obj: pd.DataFrame, store=None) -> np.ndarray:
         store["columns"] = obj.columns
         store["index"] = obj.index
 
-    return obj.to_numpy(dtype="float")
+    return obj.to_numpy()
```

Comment:
Can we move the changes in this file out of the PR?

This also risks to be breaking, we should ensure the previous behaviour is kept in the non-categorical case.

Reply:
I've removed the change from this pr. 
Re the case that is affecetd by this, it is when we pass categorical to the TTS adaptor using one of the sklearn encoders with local pooling. Since default input type is numpy, it throws an error because it tries to coerce it to float.

 - One solution would be to change the default value of `input type` param in TTS to pandas so that tests are not affected. Is this okay?
 - Another doubt I have is, why do we need two different input types in TTS ?

Reply:
can you explain which estimator fails on which concrete input, please? Sorry if I do not get the situation and assumptions in your scenario.

Reply:
> it is when we pass categorical to the TTS adaptor using one of the sklearn encoders with local pooling.

This is the case. 
Code that would fail:
```python
enc = TabularToSeriesAdaptor(LabelEncoder()))
enc.fit(df)
```
where df contains categorical data.

Reply:
why would it fail, in relation to the numpy conversion?

Reply:
oh, I see, because the default tries to convert to `numpy` internally.

In that case, I think it's a bad idea to change the conversion, something on a very core framework level.

Instead, I think a cleaner way would be to use the `pandas` mode when there are categorical features.

Given the current framework, there do not seem to be an easy way to do that, as the tag in `__init__` tells the estimtaor what to convert to, and we see the feature_kind only in `fit`.

Do you have any good ideas?

Reply:
> (one option would be to change the default internal type to pandas)

Yes, this was the idea that I had suggested in the earlier comment as well. 
I don't see an option to change the `input_type` dynamically since we do not have info about the dtypes during `__init__` . Also, I had asked earlier too, why are we having this option of selecting `input_type`? Like what is the use case with respect to users ?

I think keeping the default as pandas also makes more sense since our aim is that the user shouldn't have to make unnecessary extra specifications when using categorical data. Keeping pandas as default will allow this. And if the user happens to want to use `input_type` as `numpy`, then the expectation is that categorical is not present in the data. If it is present, then it fails since we are not supporting categorical data in numpy now.

Reply:
Changing the default `input_type` broke many other tests which depended on the previous default value, so that fix would not be feasible right now. What I've currently done is to set `pooling` to global in the test params since when pooling is global, input_type is changed to pandas in init, so the tests do not fail.

One thing I realised is that shouldn't TTS adapter be considered a composite and be ruled out for the test in the first place ?

Reply:
>  This is the only solution I see that will work for now.

But that's just a solution to get the tests pass, not for the underlying issue that default usage will break when a passing option is available - this impacts most unfortunately the dunders for pipelines, which use the default setting.

> One thing I realised is that shouldn't TTS adapter be considered a composite and be ruled out for the test in the first place ?

I suppose so! I see that `is_composite` does not pick it up, because the component is an `sklearn` estimator.

Further, even if we do not test, the underlying problem of failure with categorical `pandas` under direct usage is not solved. Perhaps it is a different problem though.

Another question - should we change the default to `"pandas"` use internally, mid- or long-term? @sktime/core-developers

Reply:
> But that's just a solution to get the tests pass, not for the underlying issue that default usage will break when a passing option is available
Yes that is correct.

But technically, we are not allowing categorical data in numpy format yet, so the failure when `input_type` is `numpy` is not deviating from the design right ? So the expectation can be that if the user wishes to use categorical, they must select `input_type` as `pandas`.

Reply:
Yes, I think that is a fair assumption.

Though stillI maintain we should not change existing contract tests or conversions to "fudge" a test pass. Can you find another way that does not "cut corners" with contracts but instead corrects the test conditions?

Reply:
Yes, I've already removed that coercion change. 

If the solution of adding the pooling type as global in the test param is not acceptable, then the other solution I see is to try and exclude this one particular case alone from the test by using an if statement before the test.

Reply:
> If the solution of adding the pooling type as global in the test param is not acceptable,

I think not, this is "fudging the tests" which removes coverage of a case that previously passed.

> then the other solution I see is to try and exclude this one particular case alone from the test by using an if statement before the test.

That might be a solution, but again we should not do an if statement that is estimator specific, e.g., "if tabular transform adapter then". To have a unified API it needs to use a property or method of the estimator only.

Reply:
I figured out a fix! I can just set the categorical support tag to False in `__init__` if the `input_type` is `numpy`. I'm kind of surprised this didn't cross my mind earlier.

Reply:
yes, that's a brilliant solution, I did not think of it - it preserves the boundaries of the various API designs and avoid making any changes to extraneous contracts. 👍